napari/_qt/utils.py

```diff
@@ -151,3 +151,45 @@ def square_pixmap(size):
     painter.drawRect(1, 1, size - 3, size - 3)
     painter.end()
     return pixmap
+
+
+def drag_with_pixmap(list_widget: QListWidget) -> QDrag:
```

Comment:
This is so nice - really makes me think we can drop the whole custom layer list widget and use a similar approach here - the only thing we need to make sure is that we can embed the thumbnails and shape icons inside the `QListWidgetItem` - that would be such a nice refactor @tlambert03 :-)

Reply:
yeah, I agree!  I feel like that little function there helped me feel like there might just be a clean path for that `LayerList` refactor.  Will think about that with all my newfound free time.